lib/cea/dtvcc_packet_builder.js

```diff
@@ -0,0 +1,169 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+goog.provide('shaka.cea.DtvccPacketBuilder');
+goog.provide('shaka.cea.DtvccPacket');
+
+goog.require('shaka.cea.Cea708Service');
+
+
+// CEA-708 DTVCC Packet Builder.
+// Builds packets based on Figure 5 CCP State Table in 5.2 of CEA-708-E.
+// Initially, there is no packet. When a DTVCC_PACKET_START payload is received,
+// a packet begins construction. The packet is considered parsed once all bytes
+// indicated in the header are read, and ignored if a new packet is started
+// before the current packet being processed is parsed.
```

Comment:
I'm having trouble with your terminology here.  Let's discuss in our scheduled meeting today and see if we can work it out.